### PR TITLE
[graph_trainer] Enable ChunkedCELoss for deepseek_v3 and qwen3

### PR DESCRIPTION
**Pull request by ydwu4**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):

* __->__ #3248

Previously ``to_graph_trainer_config`` swapped any ``ChunkedCELoss``
config to ``CrossEntropyLoss`` because graph_trainer's tracer relied
on ``param.grad`` side effects that ChunkedCELoss's manual chunked
backward didn't preserve through the FX trace.

With ``ChunkedCELoss(support_autograd_grad=True)`` (commit a) plumbing
sharded lm_head grads as explicit autograd outputs, and the SAC + bucketing
pass fixes (commit d) that prevent loss-region nodes from being
recomputed/reordered, ChunkedCELoss now traces correctly. Flip
``support_autograd_grad=True`` instead of swapping to CrossEntropyLoss.

Cleanup that follows the swap removal:

  - ``deepseek_v3_debugmodel_ep_ce_loss``, ``qwen3_debugmodel_ce_loss``,
    ``qwen3_moe_debug_ep_ce_loss``: graph_trainer-only baselines that
    existed solely so loss_compare's eager side matched graph_trainer's
    forced CE. Now obsolete; deleted. (``llama3_debugmodel_ce_loss``
    in the main models registry stays — separately used by TP/SP
    integration tests.)
  - ``test_numerics.py``: baselines for llama3, qwen3 (dense + MoE),
    and deepseek_v3 flip from ``*_ce_loss`` to the standard
    chunked-CE configs. Both eager and graph_trainer now use
    ChunkedCELoss for these comparisons.

Verified end-to-end (FSDP=4 x TP=2 [x EP=2 for MoE], deterministic
seed=42, 5 steps, bitwise loss + grad_norm match against eager):

  - graph_trainer.qwen3_debugmodel                       (dense)
  - graph_trainer.llama3_debugmodel                      (dense)
  - graph_trainer.deepseek_v3_debugmodel                 (dense)
  - graph_trainer.deepseek_v3_debugmodel_ep              (MoE)
  - graph_trainer.deepseek_v3_debugmodel_flex_attn_ep    (MoE+flex)
  - graph_trainer.qwen3_debugmodel_moe_ep                (MoE)

### CODE REVIEW

File: torchtitan/experiments/graph_trainer/configs.py [hunk 1]
```diff
@@ -131,9 +131,11 @@ def to_graph_trainer_config(
...
         d["activation_checkpoint"] = ActivationCheckpointConfig(mode="selective")
 
-    # TODO: graph_trainer doesn't yet support ChunkedCELoss
-    if isinstance(d.get("loss"), ChunkedCELoss.Config):
-        d["loss"] = CrossEntropyLoss.Config()
+    # graph_trainer's tracer requires explicit autograd outputs for lm_head
+    # params instead of relying on .grad side effects from chunk_loss.backward().
+    loss_cfg = d.get("loss")
+    if isinstance(loss_cfg, ChunkedCELoss.Config):
+        loss_cfg.support_autograd_grad = True
```
Comment (SherlockNoMad) [inline torchtitan/experiments/graph_trainer/configs.py]: update to use graphtrainer's chunkloss version.

File: torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py [hunk 2]
```diff
@@ -72,12 +70,3 @@ def graph_trainer_deepseek_v3_671b() -> GraphTrainer.Config:
     return config
 
 
-# CrossEntropyLoss baseline for graph_trainer numerics tests. graph_trainer
```
Comment (SherlockNoMad) [inline torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py:92]: just double checking, llama3 should also have the right configs

**Comment by SherlockNoMad** [comment]
test_bitwise_deterministic.py test cases should also have a variant that use chunk loss

**Review by SherlockNoMad — APPROVED** [review]
